xkcd.py

```diff
@@ -1,11 +1,56 @@
-from flask import Flask
+from flask import Flask, request, jsonify, abort
+import urllib.request, json
+from datetime import datetime
+import os
+
 app = Flask(__name__)
 
-@app.route('/comics/current')
+PORT =  5000 if os.environ.get('FLASK_PORT') is None else os.environ.get('FLASK_PORT')
+HOST = '127.0.0.1' if os.environ.get('FLASK_HOST') is None else os.environ.get('FLASK_Host')
+
+@app.route('/comics/current', methods=['GET'])
 def current_comic():
-    pass
+    current_comic_url = "http://xkcd.com/info.0.json"
+    return comic_by_url(current_comic_url)
+
+
+@app.route('/comics/<int:comic_id>', methods=['GET'])
+def comic_by_id(comic_id):
+    return comic_by_url(create_url_from_id(comic_id)), 200
+
+ 
+@app.route('/comics/many', methods=['GET'])
+def many_comics():
+    # we are assuming we don't want repetition and don't care about order of comics
+    comics_ids = set(request.args.to_dict(flat=False).get("comic_ids"))
+    return jsonify( [comic_by_url(create_url_from_id(comic_id)) for comic_id in comics_ids] )
```

Comment:
you're returning a list here (which is OK, it's according to the task). There is just tiny issue here: when you return a list somewhere, the person who reads it usually thinks that it is ordered by something.
So for lists we should keep some ordering. For dictionaries it's fine to keep random order :)